controllers/eventbus/installer/jetstream.go

```diff
@@ -498,41 +489,117 @@ func (r *jetStreamInstaller) createSecrets(ctx context.Context) error {
 	}
 
 	oldCObj := &corev1.Secret{}
-	if err := r.client.Get(ctx, client.ObjectKeyFromObject(clientAuthObj), oldCObj); err != nil {
+	if err := r.client.Get(ctx, apitypes.NamespacedName{Namespace: r.eventBus.Namespace, Name: generateJetStreamClientAuthSecretName(r.eventBus)}, oldCObj); err != nil {
 		if apierrors.IsNotFound(err) {
 			oldClientObjExisting = false
 		} else {
 			return fmt.Errorf("failed to check if nats client auth secret is existing, err: %w", err)
 		}
 	}
 
-	if oldClientObjExisting && oldServerObjExisting { // Both existing, do nothing
-		return nil
-	}
+	if !oldClientObjExisting || !oldServerObjExisting {
+		// Generate server-auth.conf file
+		encryptionKey := common.RandomString(12)
+		jsUser := common.RandomString(8)
+		jsPass := common.RandomString(16)
+		sysPassword := common.RandomString(24)
+		authTpl := template.Must(template.ParseFS(jetStremAssets, "assets/jetstream/server-auth.conf"))
+		var authTplOutput bytes.Buffer
+		if err := authTpl.Execute(&authTplOutput, struct {
+			JetStreamUser     string
+			JetStreamPassword string
+			SysPassword       string
+		}{
+			JetStreamUser:     jsUser,
+			JetStreamPassword: jsPass,
+			SysPassword:       sysPassword,
+		}); err != nil {
+			return fmt.Errorf("failed to parse nats auth template, error: %w", err)
+		}
+
+		// Generate TLS self signed certificate for Jetstream bus: includes TLS private key, certificate, and CA certificate
+		hosts := []string{}
+		hosts = append(hosts, fmt.Sprintf("%s.%s.svc.cluster.local", generateJetStreamServiceName(r.eventBus), r.eventBus.Namespace)) // todo: get an error in the log file related to this: do we need it?
+		hosts = append(hosts, fmt.Sprintf("%s.%s.svc", generateJetStreamServiceName(r.eventBus), r.eventBus.Namespace))
 
-	if oldClientObjExisting {
-		if err := r.client.Delete(ctx, oldSObj); err != nil {
-			return fmt.Errorf("failed to delete malformed nats server auth secret, err: %w", err)
+		serverKeyPEM, serverCertPEM, caCertPEM, err := tls.CreateCerts(certOrg, hosts, time.Now().Add(10*365*24*time.Hour), true, false) // expires in 10 years
+		if err != nil {
+			return err
 		}
-		r.logger.Infow("deleted malformed nats server auth secret successfully")
-	}
 
-	if oldServerObjExisting {
-		if err := r.client.Delete(ctx, oldCObj); err != nil {
-			return fmt.Errorf("failed to delete malformed nats client auth secret, err: %w", err)
+		// Generate TLS self signed certificate for Jetstream cluster nodes: includes TLS private key, certificate, and CA certificate
+		clusterNodeHosts := []string{}
+		for i := 0; i < r.eventBus.Spec.JetStream.GetReplicas(); i++ {
```

Comment:
Is it able to do "*.xxx"? Updating replicas will not regenerate the CERT.

Reply:
Good thinking. Let me see.

Reply:
Great, using a wildcard in the first position seems to work: 
`"fmt.Sprintf("*.%s.%s.svc.cluster.local", generateJetStreamServiceName(r.eventBus), r.eventBus.Namespace)"`